app/fastlane/helpers/slack.rb

```
@@ -0,0 +1,105 @@
+          if response.code == "429"
+            raise "HTTP 429 Rate limit exceeded for Slack API"
+          end
+
+          raise "Slack API failed: #{response.code} #{response.body}" unless response.is_a?(Net::HTTPSuccess)
+          json = JSON.parse(response.body)
+          raise "Slack API Error: #{json["error"]}" unless json["ok"]
+          upload_url = json["upload_url"]
+          file_id = json["file_id"]
+        end
+        [upload_url, file_id]
+      end
```

RuboCop (1.76.1)

[convention] Assignment Branch Condition size for `request_upload_url` is too high. [<11, 23, 4> 25.81/23]

(Metrics/AbcSize)